File: reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = 1
        if x < 0:
            sign = -1

        x *= sign

        ans = 0
        while x > 0:
            ans = ans*10 + x%10
            x //= 10

        if ans*sign < -2**31 or ans*sign > (2**31-1):
            return 0
        else:
            return ans*sign

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int num = x;
        boolean negative = x<0?true:false;
        if (negative) num*=-1;
        long ans = 0;

        while (num>0){
            ans*=10;
            ans+=(num%10);
            num/=10;
            if (ans>=Integer.MAX_VALUE) return 0;
        }
        return (int)(negative?-1*ans:ans);
    }
}